### PR TITLE
Feat/pressable links ios

### PR DESCRIPTION
**Pull request by teniii**
Added the pressable link functionality for iOS. 

For this, the component needs an extra prop, `linksArray`, where you should pass the links presented in the content, in the order of appearance.

### CODE REVIEW

**Comment by teniii** [comment]
https://streamable.com/u8w8nw

File: ios/RNSelectableTextView.m [hunk 1]
```diff
@@ -126,10 +127,24 @@ -(void) handleSingleTap: (UITapGestureRecognizer *) gesture
     const NSInteger location = [_backedTextInputView offsetFromPosition:beginning toPosition:selectionStart];
     const NSInteger endLocation = [_backedTextInputView offsetFromPosition:beginning toPosition:selectionEnd];
     
+    const NSInteger tapLocation = [_backedTextInputView offsetFromPosition:beginning toPosition:tapPos];
+    NSString* url;
+    NSUInteger urlAssociatedLocation, urlAssociatedLength;
+    
+    
+    for (NSDictionary* item in urlAndRange){
+        urlAssociatedLocation = [ [item objectForKey:@"location"] unsignedIntValue] ;
```
Comment (likesebiii) [inline ios/RNSelectableTextView.m]: space left here at beginning

**Review by likesebiii — APPROVED** [review]

File: ios/RNSelectableTextView.m [hunk 2]
```diff
@@ -185,6 +198,45 @@ - (void)setAttributedText:(NSAttributedString *)attributedText
...
+    // for each font-color found in the attributed string
+    [res enumerateAttribute:NSForegroundColorAttributeName inRange:NSMakeRange(0, res.length) options:0 usingBlock:^(id value, NSRange range, BOOL *stop) {
+        // if a font-color was found in the attributed string
+        if (value) {
+            UIColor *oldColor = (UIColor *)value;
+            colors = CGColorGetComponents(oldColor.CGColor);
+            
+            // the red component of a link text color is 0.4 and 0.72 depending of the theme
+            // regular text has values around 0.12 on light and 1 on dark
+            if( colors[0] > 0.4 && colors[0] < 0.8) {
```
Comment (likesebiii) [inline ios/RNSelectableTextView.m]: spacing here too

File: ios/RNSelectableTextView.m [hunk 3]
```diff
@@ -185,6 +198,45 @@ - (void)setAttributedText:(NSAttributedString *)attributedText
...
+                // make sure the linksArray won't be out of range
+                if(self.linksArray.count > urlAndRange.count){
```
Comment (likesebiii) [inline ios/RNSelectableTextView.m]: space after if and after )

File: ios/RNSelectableTextView.m [hunk 4]
```diff
@@ -185,6 +198,45 @@ - (void)setAttributedText:(NSAttributedString *)attributedText
...
+                    NSDictionary* itemToBeAdded =   @{
```
Comment (likesebiii) [inline ios/RNSelectableTextView.m]: double spacing

**Pull request merged**